packages/mui-utils/src/resolveProps/resolveProps.ts

```diff
@@ -14,36 +16,51 @@ export default function resolveProps<
 >(defaultProps: T, props: T) {
   const output = { ...props };
 
-  (Object.keys(defaultProps) as Array<keyof T>).forEach((propName) => {
-    if (propName.toString().match(/^(components|slots)$/)) {
-      output[propName] = {
-        ...(defaultProps[propName] as any),
-        ...(output[propName] as any),
-      };
-    } else if (propName.toString().match(/^(componentsProps|slotProps)$/)) {
-      const defaultSlotProps = (defaultProps[propName] || {}) as T[keyof T];
-      const slotProps = props[propName] as {} as T[keyof T];
-      output[propName] = {} as T[keyof T];
+  for (const key in defaultProps) {
+    if (Object.prototype.hasOwnProperty.call(defaultProps, key)) {
+      const propName = key as keyof T;
+
+      if (propName === 'components' || propName === 'slots') {
```

Comment:
Same for the regex name check vs string comparison. Regexes are quite efficient on modern engines but there is still some overhead: https://jsben.ch/iNyQn